### ISSUE TITLE
Adapt Renku charts to include GitLab cloud native

### ISSUE THREAD

**Issue by pameladelgado** [SwissDataScienceCenter/renku#2094]
Following the GitLab cloud native PoC deployment #515 , we should adapt our charts to include this as part of Renku.
The GitLab deployment should be as independent as possible, for instance use a separate Postgresql deployment and DB.
Appart from including the GitLab deployment as part of Renku, we need to automate a couple of things. Notably the integration with Renku's keycloak.

(the following list might be updated)
- [x] Include a GitLab deployment with Renku (verify how to merge global sections)
~- [ ] Investigate how to allow two Postgresql deployments to live within the same release/namespace~
- [x] Unify current GitLab application secrets (UI and jupyterhub).
- [x] Disable post-install GitLab and references to non cloudnative deployment.
- [x] Make a new secret to store the information for omni-auth connection.
~- [ ] Add a secret containing LFS information.~
~- [ ] Potentially the same as above for registry configuration.~
- [x] Test the new charts with a deployment.
- [ ] Document values file

**Comment by ableuler** [comment]
Thanks for putting this detailed description together. As briefly mentioned, I think there are two aspects to an "independent" vs "non-independent" GitLabs. I believe that for serious production settings it would be a good idea to deploy gitlab independent of RenkuLab. For those cases we should provide a as clear as possible list of requirements (supported GitLab versions, lfs, registry, runners, client applications, etc).
At the same time I also believe that we still want to include the option to deploy GitLab with our Renku chart. In this case using the same postgres instance seems much easier and even preferable to me, since it would make it easier to turn this one postgres instance into a higly available instance.

**Comment by pameladelgado** [comment]
Indeed, that's what I've been looking at (shared postgresql). I've updated the description, thanks for the reminder.
